### ISSUE TITLE
Older APKs do not get packaged

### ISSUE THREAD

**Issue by JustArchi** [opengapps/opengapps#97]
In today's ```open_gapps-arm-5.1-stock-20150820.zip``` package I found out that only latest versions of APKs are packaged.

This is a major problem because in above package there is only 480-based ```GmsCore```, which will greatly fail when installed on any non-480 device.

Please correct this, it's better to ship older working package rather than no package at all.

**Comment by DJAlik** [comment]
I have this GMS running on my 320-based device and it's working just fine

On Thu, Aug 20, 2015 at 2:23 PM, Łukasz Domeradzki <notifications@github.com
> wrote:

> In today's open_gapps-arm-5.1-stock-20150820.zip package I found out that
> only latest versions of APKs are packaged.
>
> This is a major problem because in above package there is only 480-based
> GmsCore, which will greatly fail when installed on any non-480 device.
>
> Please correct this, it's better to ship older working package rather than
> no package at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opengapps/opengapps/issues/97>.
>



-- 
<http://www.facebook.com/djalik>  <http://twitter.com/#!/DJAlik>
<http://www.linkedin.com/in/alexrusinov>
<http://plus.google.com/114580313768137070684>
<http://www.mixcloud.com/djalik/>
<http://psnprofiles.com/DJAlik>

**Comment by JustArchi** [comment]
It's not optimized for 320-based devices and forces Google Services to download correct GMSCore on update anyway.

**Comment by mfonville** [comment]
Hmm. This was a design choice based on the information that higher dpi versions worked without issues on lower dpis. I know that some other gapps like slim and banks did this in the past if a spexific dpi was not yet available.
The script picks the very latest version available without looking add the dpi and then adds dpi variations with matching versionname. When installed it does not get downgraded, but will receive still the proper updates, when applicable, for the correct dpi.
@JustArchi which errors appear when installing this wrong-dpi version?

**Comment by JustArchi** [comment]
Errors? None. More like potential incompatibilities and unwanted bloat, which could be avoided. If you're still wondering what I'm talking about then just compare different DPI packages of gmscore, their sizes and contents, and you'll find out. I know it's easier to package one universal version for everything but GAPPS are already heavy and overbloated, bloating them further is the last thing I want to do.

**Comment by mfonville** [comment]
I think that the mentioned bloat is mostly scaling of the resources. Which is not ideal, but also not giving trouble atm.
Maybe we should approach this problem by doing some branching with 'stable' builds that have full dpi suolpport, and some cutting-edge builds that focus on the latest available version?
@rapperskull when you are back from holiday, could you give your opinion on this?

**Comment by rapperskull** [comment]
We already addressed this issue when we added multi-DPI support. The choice was to prefer newer version over DPI correctness, even if initially I was of the other opinion. I'll explain why:
1) Installing the wrong package is not a big deal, as lower versions will be updated and resources from upper versions will be scaled. The next package flashing will correct the problem.
2) Variations come and go and to maintain a package with all DPIs we have to manually compile a list of variations, which we don't have, since we're not the developers of the apps and it should be compiled with what we currently have (what we already do packaging only available APKs). For example the -432 version of GMSCore is not available on APKMirror from a long time now. We would not be able to package our GApps if we were still waiting for that variation to appear again.
